### PR TITLE
[#10986] Show loading message below response loading spinner

### LINKED ISSUE #10986: Show loading message below response loading spinner
From PR #10960 

**Description of feature/enhancement**

If a course has many students (or in general many responses per question), add extra loading text below the spinner to inform the user that the students are being loaded (after perhaps 5s).

image

**Justification**

From @damithc 
> Provided the load operation will not time out, perhaps simply show a message below the spinner cautioning the user that this particular question will take a long time to load?

This would give user more assurance that there is progress in the loading.

**Existing similar features and their shortcomings**

Experimented with using `forkJoin` in the frontend to fetch questions in parallel in #10960: the parallel calls create too much instantaneous load at the backend and causes new instances to be spun up. The cost here is not justified.

### PR DESCRIPTION
**Pull request by nielsly**
Fixes #10986

**Outline of Solution**

We added a simple timeout which shows an additional message in all loading spinners after 5 seconds, as suggested in Issue #10986.
This only affects the front end, and does nothing to ensure the loading is still happening besides it making sure the page is not hanging.

While working on this we found #11037 did something very similar, it seems like that PR was not accepted due to it becoming inactive, but as the content was very similar, we would gladly like to hear if our fix is not good enough.

In our opinion, it could be improved by making the loading text dynamic, e.g. by adding extra periods to the end of the "Still loading..." message. This has seemed to work pretty well, though it breaks some tests when we tried it to see. 
Another issue is that just before the loading finishes the page will hang for 1-2 seconds, pausing the dynamic text, but not the spinner, this might confuse users. 
This hanging also causes the "Still loading..." message to not appear after 5 seconds, if the actual loading finishes before then. This is a minor issue as it should still show the results in 1-2 seconds as said before.

### CODE REVIEW

File: src/web/app/components/loading-spinner/loading-spinner.component.html
Comment (jasonqiu212) [inline src/web/app/components/loading-spinner/loading-spinner.component.html:4]: Instead of adding another message underneath, I think it would look better to update the original message `Loading...` to `Still loading...`.

Alternatively, as you mentioned, would it be possible to dynamically cycle the loading message (E.g. `Loading.` -> `Loading..` -> `Loading...` -> `Loading.` -> ...). I think that would look cooler and give the user more reassurance :)

**Review by jasonqiu212 — COMMENTED** [review]
Hi @nielsly, Thanks for picking up this issue!

As you mentioned, I think it would be nice to have a dynamic loading message (Maybe one that can cycle through `Loading.`, `Loading..`, and `Loading...`?) 

@zhaojj2209 @samuelfangjw I was wondering what do you guys think about this issue?
- I think the most ideal solution is to have a progress bar that actually corresponds to how much data is fetched. However, I can imagine that being quite a big issue and requires batching. 
- Thus, before we implement batching for API calls, I think this PR serves well as a reassurance to users who are impatient to wait for long loading times and can be merged in after some minor changes.
- If we do implement batch calls in the future, we can change this loading message to a real progress bar that represents progress

**Comment by samuelfangjw** [comment]
@jasonqiu212, my opinion on this:

1. This particular performance issue will likely be resolved with v9. If the updates work as intended, the current loading spinner should actually be adequate for the task. In this case there wouldn't be a need for a progress bar.
2. No harm adding your suggestion of updating the message to `still loading...`. I think this is sufficient given the circumstances, not very keen on adding more unnecessary complexity.
3. I don't think there is much value to dynamically cycle the loading message, the loading spinner already serves the purpose of hinting that the action is still being carried out.

**Comment by jasonqiu212** [comment]
@samuelfangjw Thanks for the good points raised!

@nielsly Would it be possible to update the original message `Loading...` to `Still loading...`, rather than adding a new message beneath? Thank you!

**Comment by nielsly** [comment]
I have now changed it to what @jasonqiu212 described, let me know if there are any other things that should be changed regarding this issue!

**Review by jasonqiu212 — APPROVED** [review]
LGTM! Thanks for the contribution :)

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

**Pull request merged**
